alternator/executor.cc

```diff
@@ -128,6 +128,20 @@ void executor::supplement_table_info(rjson::value& descr, const schema& schema,
 // See https://github.com/scylladb/scylla/issues/4480
 static constexpr int max_table_name_length = 222;
 
+static bool valid_table_name_chars(std::string_view name) {
+    for(auto c : name) {
+        if ((c < 'a' || c > 'z') &&
+            (c < 'A' || c > 'Z') &&
+            (c < '0' || c > '9') &&
+            c != '_' &&
+            c != '-' &&
+            c != '.') {
+            return false;
```

Comment:
If you already know how to benchmark this, I wonder whether we can do even faster by replacing this long list of comparisons by a single lookup table, i.e., a switch, something like (using https://gcc.gnu.org/onlinedocs/gcc/Case-Ranges.html, a non-standard C extensions supported by both gcc and clang):
```c++
switch(c) {
  case 'a'..'z':
  case 'A'..'Z':
  case '0'..'9':
  case '_':
  case '-':
  case '.':
      break;
  default:
      return false;
}
```

Reply:
Thanks, I'll check that version too. 

Funny how this benchmarking is useful, initially I coded it with `std::isalnum` but it turned to be 50% slower (!), I think due to locale handling. And interestingly while it was slower it still increased throughput (I suspect because it removed 4 allocations).

Reply:
I ran it, it's not conclusive, tps is similar, insns/op too. I think one'd need to microbenchmark just this func to tell the which one is faster.

Reply:
> I think one'd need to microbenchmark just this func to tell the which one is faster.

Bad idea. Microbenchmarks tell you how fast a piece of code is when you run it in a tight loop, which doesn't happen in a real workload. The drawback of a lookup table is that the table has to be read from memory, often in random order. If you are unlucky, this might completely dominate the cost of the function in a real workload, but be invisible in microbenchmarks.

Reply:
I assume that includes import of a table?

Reply:
Import involves looking up an existing table (not that the lookup is performance sensitive).

Practically the only APIs that create names are [CreateTable](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html) and CreateGlobalTable.

Reply:
> The drawback of a lookup table is that the table has to be read from memory

Yes, indeed, as I understand https://godbolt.org/z/WeYsvn1er the table version will generate ~300 bytes of rodata, seems like a lot for such a small function, but yes it's difficult to quantify as:
- microbenchmarks should account for data load, simple `for` loop will be biased as mentioned
- it's too tiny to be visible on more integration/e2e level profile (maybe I need to tweak my pprof config)

Reply:
Those 300 bytes are not a problem. I'm sure the regex is even larger.